test/macros.jl

```diff
@@ -1,9 +1,23 @@
+using AutoHashEquals
+using ReTestItems
+using Test
+
 # this file specifically tests *unit* tests for the macros: `@testitem` and `@testsetup`
 # so *not* the `runtests` functionality, which utilizes specific
 # contrived packages/testfiles
 n_passed(ts) = ts.n_passed
 n_passed(ts::ReTestItems.TestItemResult) = n_passed(ts.testset)
 
+function expand(f)
+    old = get(task_local_storage(), :__RE_TEST_RUNNING__, false)
+    try
+        task_local_storage()[:__RE_TEST_RUNNING__] = true
+        return f()
+    finally
+        task_local_storage()[:__RE_TEST_RUNNING__] = old
+    end
+end
```

Comment:
we could have used `_quote` for all these too, but i wanted to make sure the `__RE_TEST_RUNNING__` variable was being respected, so chose this approach here

maybe this should just be its own test and we should use `_quote` everywhere else?